### PR TITLE
[wip] optimize: optimize the service upgrade and downgrade logic

### PR DESCRIPTION
**Pull request by a364176773**
### Ⅰ. Describe what this PR did
optimize the service upgrade and downgrade logic

### CODE REVIEW

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 1]
```diff
@@ -234,21 +243,39 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
...
+                } catch (InterruptedException e) {
+                    break;
                 }
             }
-        }, degradeCheckPeriod, degradeCheckPeriod, TimeUnit.MILLISECONDS);
+        });
     }
 
-    private static synchronized void onDegradeCheck(boolean succeed) {
+    private synchronized void onDegradeCheck(boolean succeed) {
```
Comment (slievrly) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java:278]: Locks are acquired for each transaction request, and I recommend reducing the scale or asynchrony of the locks

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 2]
```diff
@@ -64,14 +65,14 @@
...
-    private volatile boolean disable;
+    private static volatile boolean disable;
     private static int degradeCheckPeriod;
     private static volatile boolean degradeCheck;
     private static int degradeCheckAllowTimes;
     private static volatile Integer degradeNum = 0;
     private static volatile Integer reachNum = 0;
-    private static ScheduledThreadPoolExecutor executor =
-        new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("degradeCheckWorker", 1, true));
+    private static volatile ThreadPoolExecutor executor = new ThreadPoolExecutor(0, Integer.MAX_VALUE, 60L, TimeUnit.SECONDS,
```
[the remark this answers is not in the document]
Comment (a364176773) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java:74]: ok

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 3]
```diff
@@ -234,21 +243,39 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
...
+                }
                 try {
-                    String xid = TransactionManagerHolder.get().begin(null, null, "degradeCheck", 60000);
-                    TransactionManagerHolder.get().commit(xid);
-                    onDegradeCheck(true);
-                } catch (Exception e) {
-                    onDegradeCheck(false);
+                    Thread.sleep(degradeCheckPeriod);
+                } catch (InterruptedException e) {
+                    break;
```
[the remark this answers is not in the document]
Comment (a364176773) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java:272]: > will go step out while(true) ,print some logs

this exception enters when the thread is closed  , so need break

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 4]
```diff
@@ -234,21 +243,39 @@ public void onChangeEvent(ConfigurationChangeEvent event) {
...
+                            onDegradeCheck(true);
+                        } catch (Exception e) {
+                            onDegradeCheck(false);
+                        }
+                    } else {
+                        try {
+                            this.wait();
+                        } catch (InterruptedException e) {
+                            if (LOGGER.isErrorEnabled()) {
+                                LOGGER.error("an unknown error occurred and the self - checking thread could not wait");
```
[the remark this answers is not in the document]
Comment (a364176773) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java:263]: > contains origin InterruptedException

ok

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java (hunk 1 above)
Comment (a364176773) [reply spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java:278]: > Locks are acquired for each transaction request, and I recommend reducing the scale or asynchrony of the locks

ok

**Comment by a364176773** [comment]
@zjinlei @slievrly PTAL

**Pull request closed without merging**